### ISSUE TITLE
Failed to Deploy Artifacts:.. 401 Unauthorized when releasing plugin

### ISSUE THREAD

**Issue by sairen-carbonetes** [jenkins-infra/helpdesk#3568]
### Service(s)

Others

### Summary

Here's my plugin: https://github.com/jenkinsci/jacked-plugin/

I set up my settings.xml using the documentation, I generate the content using the 
`curl -u your_user_name:your_password https://repo.jenkins-ci.org/setup/settings.xml`
from this documentation https://www.jenkins.io/doc/developer/publishing/releasing-manually/

After I added this on pom.xml
![image](https://user-images.githubusercontent.com/122254250/236968272-cb2334c0-a595-4cb0-badc-0e3e9b981604.png)

Got this error when performing `mvn release:prepare release:perform`

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.1.1:deploy (default-deploy) on project jacked: Failed to deploy artifacts: Could not transfer artifact io.jenkins.plugins:jacked:pom:v1.2.0 from/to central (https://repo.jenkins-ci.org/artifactory/releases): status code: 401, reason phrase:  (401) -> [Help 1]`

I've tried to reset my password and it performs the same error.

**Comment by sairen-carbonetes** [comment]
I tried to reset my password and generate using curl for settings.xml but I still got an error.

**Comment by dduportal** [comment]
Hi @sairen-carbonetes , you should follow the instructions at https://www.jenkins.io/doc/developer/tutorial/create/ and compare the pom.xml from archetype with yours.

TL;DR; The distribution management is handled by the "Plugin pom parent" (https://github.com/jenkinsci/jacked-plugin/blob/main/pom.xml#L13-L18) so you have to:

- Remove the `distributionManagement` from your `pom.xml` (https://github.com/jenkinsci/jacked-plugin/blob/a020650798de35f960b1131f914d9eda64380dc4/pom.xml#L5-L11)
- Ensure that you have a `~/.m2/settings.xml` set up on your environment, which content comes from the `settings.xml` of Artifactory` retrieved with the `curl` command above
- Ensure that the `id` of the `server` item with your username and encrypted password is `maven.jenkins-ci.org` (e.g. the same as the id in the parent plugin pom at https://github.com/jenkinsci/plugin-pom/blob/00ccb7a481b2665c1fa4c9382a2edadb559aa300/pom.xml#L44-L54)


If it still does not work then you have to run Maven in debug mode and share the output (taking care or anonymizing the output of course to avoid elaking your password).

As an (easier) alternative, I would suggest you to use CD: https://www.jenkins.io/doc/developer/publishing/releasing-cd/

**Comment by sairen-carbonetes** [comment]
Hi, here's my settings.xml file and where it's located, retrieve using the command
C:\Users\<user>\.m2\settings.xml
I retrieved using this command: `curl -u your_user_name:your_password https://repo.jenkins-ci.org/setup/settings.xml`


![image](https://user-images.githubusercontent.com/122254250/237024658-7dd3acd0-2143-485c-ac1a-eecb3efb2327.png)

I'm going to remove the `distributionManagement` from my pom.xml and retry the release `mvn release:prepare release:perform`

I will let you know what's the result after the changes I made. Thank you

**Comment by sairen-carbonetes** [comment]
Hi, @dduportal. I still got the same 401 error. Here's the screenshot with debug logs
![image](https://user-images.githubusercontent.com/122254250/237030597-b5ceeffc-74e7-443d-86ed-33b5179c94d1.png)

**Comment by dduportal** [comment]
Hi @sairen-carbonetes , the HTTP/401 error means that your Maven setup does not send the correct credential (or no credential at all).

As per https://github.com/jenkins-infra/helpdesk/issues/3563, it looks like you were able to release the `
 1.2.0`: do you still have issue when manually releasing?

**Comment by sairen-carbonetes** [comment]
I think I'm good for now with the issue. Thank you for helping me yesterday, you helped me a lot 🙂.  I'll let you know if I have encountered any issues. Thank you very much.
